Singularity/Singularity/Resources/EResourceType.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace Singularity.Resources
+{
+    internal enum EResourceType
```

Comment:
Enum is a nice (better!) difference to what we modeled in the Architecture, which I'm actually supporting. However, please document better why you did this!

we actually did say that we only have an interface, and for every type of resource a new class, however this would be actually unnecessary overhead we could cleanly resolve with this enum. However, please just add a comment regarding your description of why you're doing it this way and not someway else (especially so since it's different in the documentation, but even then!)

Reply:
Sure thing, will do!